log4j-to-slf4j/src/test/java/org/apache/logging/slf4j/LoggerTest.java

```diff
@@ -165,5 +166,12 @@ public void mdc() {
         assertThat(list.strList, hasSize(2));
         assertTrue("Incorrect year", list.strList.get(0).startsWith("2010"));
     }
+
+    @Test
+    public void mdcNullBacked() {
+        assertNull("Setup wrong", MDC.getCopyOfContextMap());
+        assertTrue(ThreadContext.isEmpty());
+        assertFalse(ThreadContext.containsKey("something"));
+    }
```

Comment:
I would separate these into two tests, one for each API.

Reply:
An edge case worth testing IMO is getting a null key.

Reply:
Thank you for the feedback! I split the tests and added some tests concerning null keys.